frontend/src/app/core/services/team.service.ts

```diff
@@ -4,33 +4,74 @@ import { INewTeam } from '@core/models/INewTeam';
 import { ITeam } from '@core/models/ITeam';
 import { IUpdateTeam } from '@core/models/IUpdateTeam';
 import { HttpInternalService } from '@core/services/http-internal.service';
-import { Observable } from 'rxjs';
+import { BehaviorSubject, Observable, Subject } from 'rxjs';
 
 @Injectable({
     providedIn: 'root',
 })
 export class TeamService {
     public routePrefix = '/team';
 
-    // eslint-disable-next-line no-empty-function
-    constructor(private httpService: HttpInternalService) {}
+    private emitTeamCreationSource = new Subject<ITeam>();
+
+    public teamCreationEmitted$ = this.emitTeamCreationSource.asObservable();
+
+    private emitTeamDeletionSource = new Subject<number>();
+
+    public teamDeletionEmitted$ = this.emitTeamDeletionSource.asObservable();
+
+    private currentTeamSource = new BehaviorSubject<number | undefined>(undefined);
+
+    public currentTeamEmitted$ = this.currentTeamSource.asObservable();
+
+    constructor(private httpService: HttpInternalService) {
+        this.getCurrentUserTeams()
+            .subscribe(teams => {
+                if (teams.length > 0) {
+                    this.emitCurrentTeamChange(teams[0].id);
+                }
+            });
+    }
+
+    public emitTeamCreation(newTeam: ITeam) {
+        this.emitTeamCreationSource.next(newTeam);
+    }
+
+    public emitTeamDeletion(deletedTeamId: number) {
+        this.emitTeamDeletionSource.next(deletedTeamId);
+    }
+
+    public emitCurrentTeamChange(currentTeamId: number) {
+        this.currentTeamSource.next(currentTeamId);
+    }
```

Comment:
Could we combine some of them?

Reply:
I have an idea that it can be converted to switch case, but I am not sure whether it's a good solution

Reply:
Fixed. Refactored this code by making single event for both create and delete team by user